### PR TITLE
[SPARK-13360][PYSPARK][YARN] PYSPARK_DRIVER_PYTHON and PYSPARK_PYTHON…

### PR DESCRIPTION
**Pull request by zjffdu**
… is not picked up in yarn-cluster mode

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -692,6 +692,9 @@ private[spark] class Client(
         }
         env("SPARK_JAVA_OPTS") = value
       }
+      // propagate PYSPARK_DRIVER_PYTHON and PYSPARK_PYTHON to driver in cluster mode
+      sys.env.get("PYSPARK_DRIVER_PYTHON").foreach(env("PYSPARK_DRIVER_PYTHON") = _)
+      sys.env.get("PYSPARK_PYTHON").foreach(env("PYSPARK_PYTHON") = _)
```
Comment (jerryshao) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: @zjffdu looks like [L660](https://github.com/apache/spark/blob/master/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala#L660) has the similar codes, what's the difference?

Comment (zjffdu) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: L660 is for for setting another environment variable PYTHONPATH.

**Comment by JoshRosen** [comment]
This seems fine to me, but it might be good for someone who knows more about YARN to do the final sign-off. @vanzin, could you take a quick look at this two-line PR?

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by vanzin** [comment]
LGTM, merging to master.

The way the python executable is currently propagated in Spark is a little brittle, because it requires all paths to be the same across the cluster. But I guess that setup is common enough that we don't need to worry about it now, and this change isn't necessarily making things better or worse.

**Pull request closed without merging**